app/controllers/work_packages/auto_completes_controller.rb

```diff
@@ -30,43 +30,65 @@
 require 'rack/utils'
 
 class WorkPackages::AutoCompletesController < ApplicationController
-  before_filter :find_project
-
   def index
-    @work_packages = []
-    q = params[:q].to_s
+    scope = determine_scope
+    if scope.nil?
+      render_404
+      return
+    end
 
-    if q.present?
-      query = (params[:scope] == 'all' && Setting.cross_project_work_package_relations?) ? WorkPackage : @project.work_packages
+    query_term = params[:q].to_s
+    @work_packages = []
+    # query for exact ID matches first, to make an exact match the first result of autocompletion
+    if query_term =~ /\A\d+\z/
+      @work_packages |= scope.visible.find_all_by_id(query_term.to_i)
+    end
 
-      @work_packages |= query.visible.find_all_by_id(q.to_i) if q =~ /\A\d+\z/
+    sql_query = ["LOWER(#{WorkPackage.table_name}.subject) LIKE :q OR
+                  CAST(#{WorkPackage.table_name}.id AS CHAR(13)) LIKE :q",
+                 { q: "%#{query_term.downcase}%" }]
 
-      @work_packages |= query.visible.find(:all,
-                                           limit: 10,
-                                           order: "#{WorkPackage.table_name}.id ASC",
-                                           conditions: ["LOWER(#{WorkPackage.table_name}.subject) LIKE :q OR CAST(#{WorkPackage.table_name}.id AS CHAR(13)) LIKE :q", { q: "%#{q.downcase}%" }])
-    end
+    @work_packages |= scope.visible
+                           .where(sql_query)
+                           .order(id: :asc)
```

Comment:
http://apidock.com/rails/ActiveRecord/QueryMethods/order#1542-Order-with-hash-parameters-only-in-ActiveRecord-4-